src/averageAges.js

```diff
@@ -1,64 +1,45 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const totalLengthMan = people.filter(({ sex, died }) =>
+    sex === 'm' && (century ? Math.ceil(died / 100) === century : true));
+
+  const totalAgeMan = totalLengthMan.reduce((sum, man) =>
+    sum + (man.died - man.born), 0);
+
+  return totalAgeMan / totalLengthMan.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterWoman = people.filter(({ sex, name }) =>
```

Comment:
You store the filtering result in this variable, so:
```suggestion
  const filteredWomen = people.filter(({ sex, name }) =>
```
woman - жінка, women - жінки